File: labs/l01p04/main.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    for (string line; getline(cin, line);)
    {
        string buff;
        for (auto c : line)
        {
            buff += tolower(c);
        }

        string temp = buff;
        reverse(temp.begin(), temp.end());

        cout << (buff == temp ? "palindrome\n" : "not a palindrome\n");
    }
}

File: labs/l01p04/Main.java
import java.util.*;

public class Main {
    public static void main(String []args) {
        Scanner s = new Scanner(System.in);

        while (s.hasNextLine()) {
            String line = s.nextLine();

            StringBuilder builder = new StringBuilder();

            for (char c : line.toCharArray()) {
                if(Character.isLetter(c)) {
                    builder.append(Character.toLowerCase(c));
                }
            }

            StringBuilder temp = new StringBuilder(builder);
            temp.reverse();

            System.out.println(builder.toString().equals(temp.toString()) ? "palindrome" : "not a palindrome");
        }
    }
}